rascal-lsp/src/main/rascal/library/demo/lang/pico/LanguageServer.rsc

```diff
@@ -58,9 +67,93 @@ set[LanguageService] picoLanguageServer(bool allowRecovery) = {
     inlayHint(picoInlayHintService),
     definition(picoDefinitionService),
     codeAction(picoCodeActionService),
-    selectionRange(picoSelectionRangeService)
+    selectionRange(picoSelectionRangeService),
+    formatting(picoFormattingService)
 };
 
+list[TextEdit] picoFormattingService(Tree input, loc range, FormattingOptions opts) {
```

Comment:
I think the implmentation of this should be moved out side of the `LanguageServer` file, as it's quite big, and is just a separate formatter. It might be better if we could move this to rascal std lib, where the pico example lives?